src/dsp_tools/utils/connection_live.py

```diff
@@ -330,7 +333,8 @@ def put(
                 headers=headers,
                 # if data is not encoded as bytes, issues can occur with non-ASCII characters,
                 # where the content-length of the request will turn out to be different from the actual length
-                data=jsondata.encode("utf-8") if jsondata else None,
+                data=json.dumps(jsondata, cls=SetEncoder).encode("utf-8") if jsondata else None,
```

Comment:
why do we need this `SetEncoder`?

Reply:
We need it to tell the json package how to serialize objects of type `set`, `Context`, and `OntoIri`. By the way, that's also the reason why we have to do the serialization ourselves, by hand. I don't see a possibility to pass a custom Encoder to the requests package.

Reply:
we shouldn't pass sets to any json encoder, we should convert those to lists; and we absolutely should not pass custom types to a json encoder - but ok... then that's a slightly bigger refactoring, it seems.